Resources addon to dynamically add/remove css

nm123github opened the pull request:
Issue: I think there might be use-cases where folks want to quickly hop between different css resources at runtime. Here is an addon that will help with that.

## What I did
![demo](https://user-images.githubusercontent.com/8527474/47689624-d9820b80-dba7-11e8-9290-9a69a45b67d5.gif)

Commit:
rudimentary resources addon to dynamically add/remove css

Commit:
revert changes to official-storybook/config.js

Commit:
no need for two react panels

ndelangen committed:
Merge branch 'master' into pr/nm123github/4152

# Conflicts:
#	examples/official-storybook/package.json

Hypnosphi committed:
Merge branch 'master' into addon-resources

Commit:
add resources plugin complete refactor

Commit:
Merge branch 'master' of github.com:storybooks/storybook into addon-resources

Commit:
add styles to ace-editor, remove node_modules for resources addon

Commit:
Merge branch 'addon-resources' of github.com:nm123github/storybook into addon-resources

Commit:
revert unnecessary line break to manager-head

Commit:
add README, fix component style, add disable option

Commit:
update ADDONS_SUPPORT.md for resources addon

Commit:
Merge branch 'master' of github.com:storybooks/storybook into addon-resources

Commit:
add resources __snapshots__

Commit:
Merge branch 'master' of github.com:storybooks/storybook into addon-resources

Commit:
Merge branch 'master' of github.com:storybooks/storybook into addon-resources

Commit:
regenerate yarn.lock

Commit:
regenerate yarn.lock from scratch

nm123github committed:
Merge branch 'master' into addon-resources

Commit:
revert yarn.lock update

Commit:
Merge branch 'addon-resources' of github.com:nm123github/storybook into addon-resources

Commit:
create cssresources addon

Commit:
Merge branch 'master' of github.com:storybooks/storybook into addon-cssresources

Commit:
updated readme's and storyshot

Commit:
update demo gif

Commit:
update readme.md

nm123github commented:
This might seem similar to recent [resources-addon PR](https://github.com/storybooks/storybook/pull/4152) but is slightly different. This one allows you to switch between css files/themes predefined at compile time. This was a usecase that folks had asked and might be a good idea to deliver just that in this addon.

The other addon (which is more of a codeinjector-addon) will be a different PR, (most likely will update the previous one).

Commit:
update yarn.lock

nm123github committed:
Merge branch 'master' into addon-cssresources

Commit:
update yarn.lock

Commit:
update to latest yarn.lock

nm123github committed:
Merge branch 'next' into addon-cssresources

igor-dv committed:
Merge pull request #4750 from ximenean/patch-1

Add storysource to addon gallery

ndelangen committed:
Merge pull request #4712 from mrmckeb/feature/custom-react-scripts

Add support for custom `react-scripts` packages

igor-dv committed:
Merge pull request #4706 from storybooks/node-path-loading-fix

Fix NODE_PATH is not being loaded from .env

shilman committed:
4.0.5 changelog

shilman committed:
v4.0.5

shilman committed:
4.0.6 changelog

shilman committed:
v4.0.6

igor-dv commented on addons/cssresources/src/CssResourcePanel.js:
```diff
@@ -0,0 +1,155 @@
+import styled from '@emotion/styled';
+import 'pretty-checkbox/dist/pretty-checkbox.css';
+
+import css from 'react-syntax-highlighter/languages/prism/css';
```
There some performance improvement @Hypnosphi did in storysource addon for the `react-syntax-highlighter`. Take a look there

igor-dv commented on addons/cssresources/src/CssResourcePanel.js:
```diff
@@ -0,0 +1,155 @@
+                  onClick={this.handleChange.bind(this, i)}
+                />
+                <div className="state p-success" style={{ 'font-size': '14px' }}>
+                  <svg className="svg svg-icon" viewBox="0 0 20 20">
```
Can you extract this svg to a separate component?

igor-dv commented on addons/cssresources/src/index.js:
```diff
@@ -0,0 +1,23 @@
+import addons, { makeDecorator } from '@storybook/addons';
+
+export const withCssresources = makeDecorator({
```
withCssresources => withCssResources

shilman committed:
Merge pull request #4767 from storybooks/angular-expose-force-rerender

Add missing "forceReRender" and "addParameters"

igor-dv committed:
Merge pull request #4785 from dandean/patch-2

Fix variable reference in docs: story → getStory

shilman committed:
Upgrade pr-log to 3.0 + storybook-experimental

shilman committed:
4.0.7 changelog

shilman committed:
v4.0.7

shilman committed:
4.0.7 changelog

shilman committed:
4.0.7 changelog feature area

chris-lock committed:
Fixing webpack version

Looks like this may have been missed in the [migration](https://github.com/storybooks/storybook/blob/next/MIGRATION.md#webpack-4).

shilman committed:
Upgrade pr-log labels format

shilman committed:
Merge pull request #4795 from chris-lock/patch-1

Fixing webpack version

ndelangen committed:
Merge pull request #4800 from tobiasljungstrom/next

Update the Starter Guide for React

ndelangen committed:
Merge pull request #4779 from geoffreyd/patch-1

Documentation for setting up Ember has a syntax error.

ndelangen committed:
Merge pull request #4804 from storybooks/patch-discord-link

Fix invitation URL discord

igor-dv committed:
Merge pull request #4773 from storybooks/vue-knobs-optimize-on-force-render

Vue / Knobs - optimize on force render

shilman committed:
4.0.8 changelog

shilman committed:
v4.0.8

Commit:
Docs: Fix internal deadlink

shilman committed:
Merge pull request #4830 from aliceyoung9/patch-1

little documentation fix

Commit:
Merge branch 'master' of github.com:storybooks/storybook into addon-cssresources

Commit:
Merge branch 'addon-cssresources' of github.com:nm123github/storybook into addon-cssresources

Commit:
Merge branch 'next' of github.com:storybooks/storybook into addon-cssresources

Commit:
revert unrelated changes

Commit:
revert change to guide-react doc

Commit:
remove pretty-checkbox

Commit:
circleci failing, update yarn.lock

Commit:
revert yarn.lock

Commit:
Merge remote-tracking branch 'upstream/next' into addon-cssresources

Commit:
merge latest next, revert yarnlock

igor-dv committed:
Fix yarn.lock

Commit:
update to 4.1.0-alpha.8

igor-dv committed:
Fix snapshots
